src/hooks/useBadgeCount.ts

```diff
@@ -0,0 +1,15 @@
+import { size } from "lodash";
+import { useInitialisedDeskproAppClient } from "@deskpro/app-sdk";
+import type { DeviceType } from "../types";
+
+const useBadgeCount = (items: DeviceType[]) => {
+  useInitialisedDeskproAppClient((client) => {
+    if (!Array.isArray(items)) {
+      return;
+    }
+
+    client.setBadgeCount(size(items));
```

Comment:
This can be `client.setBadgeCount(items.length);`. No need for 3rd party libraries as we're dealing with native/normal arrays.

Reply:
We have an approach to use `lodash`. If you don't want to use `lodash`, don't use `lodash` in your apps

Reply:
Where is that documented/agreed? Maybe one for a retro because using `lodash` everywhere when it adds no value of native functions seems like a questionable life choice.

Reply:
It's a good topic for a retro discussion.

Reply:
As discussed in Slack chat, I recommend we change this whilst it's fresh in our heads. lodash has no value over `.length` here so we should swap it out.

That said, it's probably a good idea to get a 3rd party to have a fresh lookie at it. Thoughts @AshleyDawson

Reply:
My opinion is that we should probably lean on core language features over libraries, in this case I think `array.length` is more appropriate. That said, lodash can be useful for some things, and in this case it's important to only only add the specific lodash function as I'm not sure that we currently tree-shake correctly.

We've leaned on lodash in the past as it's been convenient if already installed - but I strongly suggest that we don't include the entire lodash library going forward... the only thing it will do is increase the bundle size.

tldr; I think best practice should be: stick to core language features as much as possible and include libraries where absolutely necessary.

Reply:
> We've leaned on lodash in the past as it's been convenient if already installed - but I strongly suggest that we don't include the entire lodash library going forward... the only thing it will do is increase the bundle size.

I checked, and using individual `lodash` modules like `lodash.size`, `lodash.debounce`, etc., includes all the internal helper functions of `lodash`. If multiple individual modules are installed, they end up being larger than the standalone `lodash` package. 

At that time with David P., we decided that if you are using only one `lodash` function, it's better to install the individual module. If you are using two or more functions, then it's best to install the whole `lodash` package.

Reply:
changed to `client.setBadgeCount(items?.length ?? 0);`

Reply:
@zpawn I propose that we always install individual lodash functions

Reply:
@AshleyDawson if we use one function - better install one module. But if we use more than one function - better use `lodash` package

Reply:
Personally, I don't think we need any of `lodash` from what I've seen. `get(a, ["b","c"])` can just be `a?.b?.c;`, `map` is already in JS/TS and as we're mentioning here, `size` can be replaced `foo.length`. To me, it looks like `lodash` is an extra thing to learn which replaces/reimplements native features from the language itself but is another vector for supply chain attacks and more bloating/loading.

With proper typings and linting (e.g. `@typescript-eslint/strict-type-checked`) we probably have everything we need to build safe, performant software efficiently.

Reply:
@HappyPaul55 it doesn’t seem to be a critical moment. It would be great if we tried to keep the balance between criticism and wasting time.